config/src/test/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/resource/OAuth2ResourceServerConfigurerTests.java

```diff
@@ -668,6 +670,23 @@ public void requestWhenBearerTokenResolverAllowsRequestBodyAndRequestContainsTwo
 		// @formatter:on
 	}
 
+	@Test
+	public void requestWhenBearerTokenResolverAllowsRequestBodyAndRequestContainsTwoTokensInRequestBodyThenInvalidRequest()
+			throws Exception {
+		this.spring.register(AllowBearerTokenInRequestBodyConfig.class, JwtDecoderConfig.class, BasicController.class)
+				.autowire();
+		JwtDecoder decoder = this.spring.getContext().getBean(JwtDecoder.class);
+		given(decoder.decode(anyString())).willReturn(JWT);
+		// @formatter:off
+		MockHttpServletRequestBuilder request = post("/authenticated")
```

Comment:
With the below suggestion, you may need to add a `Content-Type` header to this test.